### PR TITLE
Add a SNABB_RSS_HASH_KEY env variable for testing

### PR DESCRIPTION
**Pull request by takikawa**
Makes intel_mp tests more deterministic to avoid occasional test failures in CI.

### CODE REVIEW

File: src/apps/intel_mp/intel_mp.lua [hunk 1]
```diff
@@ -31,6 +31,10 @@ local S           = require("syscall")
 local CallbackAlarm = alarms.CallbackAlarm
 local transmit, receive, empty = link.transmit, link.receive, link.empty
 
+-- This environment variable is for use in tests, where a random hash
+-- key is often undesirable (e.g., to ensure all queues see traffic in a test)
+local rss_hash_key = tonumber(lib.getenv("SNABB_RSS_HASH_KEY"))
```
Comment (wingo) [inline src/apps/intel_mp/intel_mp.lua]: We already have an environment variable for making deterministic random numbers for tests: SNABB_RANDOM_SEED.

File: src/apps/intel_mp/intel_mp.lua [hunk 2]
```diff
@@ -729,7 +733,11 @@ function Intel:rss_enable ()
 end
 function Intel:rss_key ()
    for i=0,9,1 do
-      self.r.RSSRK[i](math.random(2^32))
+      if rss_hash_key then
+         self.r.RSSRK[i](rss_hash_key)
+      else
+         self.r.RSSRK[i](math.random(2^32))
```
Comment (wingo) [inline src/apps/intel_mp/intel_mp.lua]: If we were to keep this, we should reduce this to one line (`rss_hash_key or math.random(2^32)`.  But I think instead we should make the RSS tests set SNABB_RANDOM_SEED.  WDYT?

Comment (takikawa) [reply src/apps/intel_mp/intel_mp.lua]: Yeah that sounds reasonable, I will try the change out.

**Comment by takikawa** [comment]
#1326 is a simpler solution for this, so I'll close this PR.

**Pull request closed without merging**